roles/openshift_repos/tasks/main.yaml

```diff
@@ -1,13 +1,20 @@
 ---
+- set_fact:
+    deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+    openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
```

Comment:
Instead of using set_fact here, you should be able to provide a default for openshift_deployment_type:
roles/openshift_repos/defaults/main.yaml:
```
openshift_deployment_type: "{{ deployment_type | default(None) }}"
```
If openshift_deployment_type is defined elsewhere it will override the value set here.